### DISCUSSION TITLE
Split package into Dart-only package and Flutter-dependent package?

### DISCUSSION THREAD

**Discussion by MarcelGarus** [singerdmx/flutter-quill#438] [Feature Request]
We're planning to use this package for our app platform. Our frontend uses Flutter and the backend Dart. It would be nice if we could use the data structures provided by this package on both systems, but the entire package is Flutter-dependent. Would it be possible to split this package into the following two parts?

* A Dart-only package containing the data structures. This package can be used both in Flutter- and non-Flutter applications, i.e., also on servers or desktop Dart-only machines.
* A Flutter-dependent package containing the rest (all of the UI-related stuff). This can only be used by Flutter applications.

I see two ways to go ahead with this proposal:

* **Depend on one of the other available packages that contain an abstraction for Delta data classes.** This would basically remove all the Delta data classes from this package and instead use the data classes of the other package.
  While this would increase interoperability in the Dart ecosystem, it also presents a breaking change to the users of this package – if they used the data classes exported by it, they now have to migrate to the newly chosen package. Of course, a breaking change would also enable cleaning up some legacy code that this package contains, like the verbose `export` files.
  For reference, here are the most prominent Dart packages that provide Delta classes:

  | Name                                                                            | Mature |   Delta classes   | Editor | Framework | Platforms | Notes                      |
  | :------------------------------------------------------------------------------ | :----: | :---------------: | :----: | :-------: | :-------: | :------------------------- |
  | [quill_delta](https://pub.dev/packages/quill_delta/versions/3.0.0-nullsafety.2) |   ✔️    |         ✔️         |   ❌    |   Dart    |    all    |                            |
  | [quill_delta_plus](https://pub.dev/packages/quill_delta_plus)                            |   ❌    |         ✔️         |   ❌    |   Dart    |    all    | Old version of quill_delta |
  | [quill_delta_dart](https://pub.dev/packages/quill_delta_dart)                            |   ✔️    |         ✔️         |   ❌    |   Dart    |    all    | fork of quill_delta        |
  | [quill_format](https://pub.dev/packages/quill_format)                                    |   ❌    |         ✔️         |   ❌    |   Dart    |    all    | Repo missing               |
  | [delta_markdown](https://pub.dev/packages/delta_markdown)                                |   ❌    | via flutter_quill |   ❌    |  Flutter  |    all    |                            |
  | [quill_markdown](https://pub.dev/packages/quill_markdown)                                |   ❌    |         ✔️         |   ❌    |  Flutter  |    all    |                            |
  | [flutter_quill](https://pub.dev/packages/flutter_quill)                                  |   ✔️    |         ✔️         |   ✔️    |  Flutter  |  mobile   |                            |
  | [angular_quill](https://pub.dev/packages/angular_quill)                                  |   ❌    |         ✔️         |   ✔️    |  Angular  |    web    |                            |

  <sup>(thanks to @JonasWanke for creating this table)</sup>

* **Just split this package into a Dart-only package and a Flutter-dependent package.** The package name `dart_quill` is still free, so a consistent naming scheme for the package is possible to achieve. The Flutter package could still export all the classes from the Dart package, so this is a non-breaking change.

What do you think?

**Comment by singerdmx** [comment]
`Just split this package into a Dart-only package and a Flutter-dependent package. The package name dart_quill is still free, so a consistent naming scheme for the package is possible to achieve. The Flutter package could still export all the classes from the Dart package, so this is a non-breaking change.`

Sure. I only want to maintain one repo. So that means you can maintain and publish the second repo for dartonly

**Reply by singerdmx** [reply]
then you can submit a PR to make this repo take a dependency on your package

**Reply by singerdmx** [reply]
Also bump the major version with your change
